datafusion/expr/src/type_coercion/binary.rs

```diff
@@ -1034,29 +1033,50 @@ fn is_time_with_valid_unit(datatype: DataType) -> bool {
     )
 }
 
+/// Timezone coercion is handled by the following rules:
+/// - If only one has a timezone, coerce the other to match
+/// - If both have a timezone, coerce to the left type
+/// - "UTC" and "+00:00" are considered equivalent
+fn temporal_coercion_nonstrict_timezone(
+    lhs_type: &DataType,
+    rhs_type: &DataType,
+) -> Option<DataType> {
+    use arrow::datatypes::DataType::*;
+
+    match (lhs_type, rhs_type) {
+        (Timestamp(lhs_unit, lhs_tz), Timestamp(rhs_unit, rhs_tz)) => {
+            let tz = match (lhs_tz, rhs_tz) {
+                (Some(lhs_tz), Some(rhs_tz)) => {
+                    match (lhs_tz.as_ref(), rhs_tz.as_ref()) {
+                        // UTC and "+00:00" are the same by definition. Most other timezones
+                        // do not have a 1-1 mapping between timezone and an offset from UTC
+                        ("UTC", "+00:00") | ("+00:00", "UTC") => Some(Arc::clone(lhs_tz)),
```

Comment:
since this case also returns the same value as the case below (`lhs_tz`), what value do we get special casing / checking for UTC ?

As in why not just 
```rust
// If both timestamps have a timezone, use the left hand timezone
(Some(lhs_tz), Some(rhs_tz)) => Some(Arc::clone(lhs_tz)),
```